noxfile causes mypy error

When mypy is set to warn on unreachable code, then upgrading nox from 2020.12.31 to 2021.6.12 causes a "Statement is unreachable" error on line 47
```
noxfile.py: note: In function "activate_virtualenv_in_precommit_hooks":
noxfile.py:47:9: error: Statement is unreachable  [unreachable]
            return
```

These are the culprit lines:
https://github.com/cjolowicz/cookiecutter-hypermodern-python/blob/735cf236cf914369136cdec4b8e8a85b021cd707/%7B%7Bcookiecutter.project_name%7D%7D/noxfile.py#L46-L47

This is my `mypy.ini`:
```
[mypy]
check_untyped_defs = True
disallow_any_generics = True
disallow_incomplete_defs = True
disallow_subclassing_any = True
disallow_untyped_calls = True
disallow_untyped_decorators = True
disallow_untyped_defs = True
no_implicit_optional = True
no_implicit_reexport = True
pretty = True
show_column_numbers = True
show_error_codes = True
show_error_context = True
strict_equality = True
warn_redundant_casts = True
warn_return_any = True
warn_unreachable = True
warn_unused_configs = True
warn_unused_ignores = True
```

--- a/docs/guide.rst
+++ b/docs/guide.rst
@@ -1980,6 +1980,7 @@ The |HPC| enables several configuration options which are off by default.
 The following options are enabled for strictness and enhanced output:
 
 - :option:`strict <mypy --strict>`
+- :option:`warn_unreachable <mypy --warn-unreachable>`
 - :option:`pretty <mypy --pretty>`
 - :option:`show_column_numbers <mypy --show-column-numbers>`
 - :option:`show_error_codes <mypy --show-error-codes>`

--- a/{{cookiecutter.project_name}}/noxfile.py
+++ b/{{cookiecutter.project_name}}/noxfile.py
@@ -43,8 +43,7 @@ def activate_virtualenv_in_precommit_hooks(session: Session) -> None:
     Args:
         session: The Session object.
     """
-    if session.bin is None:
-        return
+    assert session.bin is not None  # noqa: S101
 
     virtualenv = session.env.get("VIRTUAL_ENV")
     if virtualenv is None:

--- a/{{cookiecutter.project_name}}/pyproject.toml
+++ b/{{cookiecutter.project_name}}/pyproject.toml
@@ -68,6 +68,7 @@ fail_under = 100
 
 [tool.mypy]
 strict = true
+warn_unreachable = true
 pretty = true
 show_column_numbers = true
 show_error_codes = true

